ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -824,6 +824,8 @@ public CompletableFuture<RaftClientReply> submitClientRequestAsync(
 
     if (request.is(TypeCase.STALEREAD)) {
       replyFuture = staleReadAsync(request);
+    } else if (request.is(TypeCase.READ)) {
+      replyFuture = readAsync(request);
```

Comment:
When  `readOption == RaftServerConfigKeys.Read.Option.DEFAULT`, it needs to check Leader.  BTW, we should validate readOption in the constructor.

Reply:
done